rnd/autogpt_server/autogpt_server/server/rest_api.py

```diff
@@ -58,25 +42,22 @@ def get_user_id(payload: dict = Depends(auth_middleware)) -> str:
 
 
 class AgentServer(AppService):
-    event_queue: asyncio.Queue[ExecutionResult] = asyncio.Queue()
-    manager = ConnectionManager()
     mutex = KeyedMutex()
     use_db = False
     _test_dependency_overrides = {}
 
-    async def event_broadcaster(self):
-        while True:
-            event: ExecutionResult = await self.event_queue.get()
-            await self.manager.send_execution_result(event)
+    def __init__(self, event_queue: AsyncEventQueue | None = None):
+        self.event_queue = event_queue or AsyncRedisEventQueue()
 
     @asynccontextmanager
     async def lifespan(self, _: FastAPI):
+        self.run_and_wait(self.event_queue.connect())
```

Comment:
This is also called [here](https://github.com/Significant-Gravitas/AutoGPT/pull/7899/files#diff-0688cfa91a411a53076603b85abc3541971cd2de1d2812b4f467ddfed1eea443R75) on the `AppService`. Are they both needed?